source/DG.HostApp/Services/ClusterConfigActualizer.cs

```diff
@@ -0,0 +1,120 @@
+                                continue;
+                            }
+
+                            // case when amount of instance was changed
+                            int nodeInstanceAmountDiference = this.InstanceDifferenceCalculator(clusterInstanceDescription, currentHost, instance.Value.Count);
+
+                            if (nodeInstanceAmountDiference > 0)
+                            {
+                                this.applicationOrchestrator.BuildInstance(clusterInstanceDescription.Type, clusterInstanceDescription.Name, clusterInstanceDescription.Settings?.ToString(), nodeInstanceAmountDiference);
+                            }
+                            else if (nodeInstanceAmountDiference < 0)
+                            {
+                                this.applicationOrchestrator.RemoveInstance(clusterInstanceDescription.Type, clusterInstanceDescription.Name, Math.Abs(nodeInstanceAmountDiference));
+                            }
+                        }
+
+                        // starting the required applications
+                        foreach (var application in applicationInstances)
+                        {
+                            var isInstanceExist = instances.ContainsKey(ApplicationExtensions.ConstructUniqueId(application.Type, application.Name));
+                            var isHostedonThisNode = application.PlacementPolicies.First(x => x == currentHost.Name).Any();
+
+                            if (isHostedonThisNode && !isInstanceExist)
+                            {
+                                int nodeInstanceAmount = this.InstanceDifferenceCalculator(application, currentHost, 0);
+
+                                this.applicationOrchestrator.BuildInstance(application.Type, application.Name, application.Settings?.ToString(), nodeInstanceAmount);
+                            }
+                        }
+
+                        Thread.Sleep(5000);
+                    }
+                }, cancellationToken);
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            return Task.Run(() => { Console.WriteLine("ServiceWatcher stopped"); }, cancellationToken);
+        }
+
+        private int InstanceDifferenceCalculator(ApplicationInstance applicationInstance, Core.Model.ClusterConfig.Host currnetHost, int existingInstanceAmount)
+        {
+            int.TryParse(applicationInstance.Count, out int count);
+            int nodeInstanceAmountDifference = (count / applicationInstance.PlacementPolicies.Count) - existingInstanceAmount;
+
+            var hostPlacementPosition = applicationInstance.PlacementPolicies.IndexOf(currnetHost.Name) + 1;
+            var remaindedInstances = count % applicationInstance.PlacementPolicies.Count;
+
+            if (remaindedInstances - hostPlacementPosition >= 0)
+            {
+                nodeInstanceAmountDifference++;
+            }
+
+            return nodeInstanceAmountDifference;
```

Comment:
I think we should change calculating logic this way:
int nodeInstanceAmountDifference = (count - existingInstanceAmount) / applicationInstance.PlacementPolicies.Count;
var remaindedInstances = (count - existingInstanceAmount) % applicationInstance.PlacementPolicies.Count;

I don't get the idea of getting the current Host index and using it in the calculation.
Maybe it would be easier just to sum up nodeInstanceAmountDifference and remaindedInstances.

If we have another nodes that host such kind of application instances and we want to spread the remaining instances via nodes, than we should have some additional logic, and maybe in future we should think about maximum loading value for nodes.

Reply:
We have priority in instance distribution between hosts. If host index is lower than remiandedInstances we need to add another instance:

![DivisionExplanation](https://user-images.githubusercontent.com/36800689/82826460-dfd88680-9eb5-11ea-95b3-b9cba3917573.png)

Reply:
That's a nice diagram, got it!